### PR TITLE
refactor: shared wallets related naming improvements

### PR DESCRIPTION
**Pull request by szymonmaslowski**
# Checklist

- [ ] JIRA - \<link>
- [ ] Proper tests implemented
- [ ] Screenshots added.

---

## Proposed solution

- shared wallets related naming improvements
- removed circular dependency of core/transaction details components

### CODE REVIEW

**Comment by pczeglik-iohk** [comment]
# Allure Report
[`allure-report-publisher`](https://github.com/andrcuns/allure-report-publisher) generated test report!



**smokeTests**: ✅ [test report](https://lace-qa:8PP5wBaDV6UoXj@dq4ajm5i5q7bz.cloudfront.net/smoke/linux/chrome/6222/9939698142/index.html) for [4560bcf2](https://github.com/input-output-hk/lace/pull/1290/commits/4560bcf2d735f1505ee738394f4f9443fbe56f61)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 33     | 0      | 0       | 0     | 33    | ✅     |

**Review by greatertomi — APPROVED** [review]

File: apps/browser-extension-wallet/src/views/browser-view/features/shared-wallet/SharedWallet.tsx [hunk 1]
```diff
@@ -64,11 +63,11 @@ export const SharedWallet = (): JSX.Element => {
     });
   };
 
-  const generateKeys = async (enteredPassword: string) => {
-    if (sharedKey) return sharedKey;
-    const sharedWalletKey = await generateSharedWalletKey(enteredPassword, cardanoWallet.source.wallet.walletId);
-    setSharedKey(sharedWalletKey);
-    return sharedWalletKey;
+  const generateKey = async (enteredPassword: string) => {
```
Comment (danielmain) [inline apps/browser-extension-wallet/src/views/browser-view/features/shared-wallet/SharedWallet.tsx:66]: Why returning the key and setting it using setSharedWalletKey?

File: apps/browser-extension-wallet/src/views/browser-view/features/shared-wallet/SharedWallet.tsx [hunk 2]
```diff
@@ -83,15 +82,15 @@ export const SharedWallet = (): JSX.Element => {
           exact
           path={walletRoutePaths.sharedWallet.generateKeys}
           render={() => (
-            <GenerateSharedKeysFlow
+            <GenerateSharedWalletKeyFlow
               activeWalletName={walletInfo?.name || ''}
               activeWalletType={activeWalletType}
-              generateKeys={generateKeys}
+              generateKey={generateKey}
```
Comment (danielmain) [inline apps/browser-extension-wallet/src/views/browser-view/features/shared-wallet/SharedWallet.tsx:88]: what about generateSharedKey or using the technical name xpub?

**Review by danielmain — CHANGES_REQUESTED** [review]
Tinny suggestion left

**Review by danielmain — APPROVED** [review]

**Pull request merged**